website/blog/2025-04-15-migrate-mongodb-data-ferretdb-dsync.md

```diff
@@ -0,0 +1,93 @@
+---
+slug: migrate-mongodb-data-ferretdb-dsync
+title: 'Migrating MongoDB Data to FerretDB with dsync'
+authors: [alex]
+description: >
+  Learn how to migrate your MongoDB data to FerretDB using dsync — with zero downtime and no data loss.
+image: /img/blog/ferretdb-migration-dsync.jpg
+tags: [open source, sspl, document databases, community]
+---
+
+![Migrating MongoDB Data to FerretDB with dsync](/img/blog/ferretdb-migration-dsync.jpg)
+
+If you're looking to migrate data from MongoDB to FerretDB, ensuring a smooth transition is crucial.
+
+<!--truncate-->
+
+Traditional migration workflows often rely on static dumps and manual restoration steps, which can often lead to complications.
+Skipped collections.
+Metadata mismatches.
+Data loss.
+And worst of all – no clue where things went wrong.
+
+[dsync](https://github.com/adiom-data/dsync/) by [adiom](https://adiom.com/) is a tool that connects directly to both the source and destination of MongoDB-compatible services and streams data in real time.
+It handles both the initial sync and live replication, continuously monitoring for any changes and updating the destination database accordingly.
+Using dsync, you can migrate your data from MongoDB to FerretDB.
+
+In this post, you'll learn to use dsync to migrate the data from MongoDB to a running FerretDB instance.
+
+## Prerequisites
+
+Make sure to have the following ready before you start:
+
+- [Running FerretDB instance](https://docs.ferretdb.io/installation/ferretdb/).
+- Running MongoDB instance (local or remote).
+
+## Download dsync
+
+Ensure to download the latest release of dsync from the [GitHub Releases page](https://github.com/adiom-data/dsync/releases/latest).
+For Mac users, you may need to configure a security exception to execute the binary by [following these steps](https://support.apple.com/en-ca/guide/mac-help/mh40616/mac).
+
+Alternatively, you can build dsync from the source code.
+
+```sh
+git clone https://github.com/adiom-data/dsync.git
+cd dsync
+go build
+```
+
+## Run dsync to migrate into FerretDB
+
+To migrate data from your local MongoDB instance to FerretDB, you need to specify the source and destination connection strings.
+
+Let's say:
+
+- MongoDB is running locally at `mongodb://localhost:27018/`
+- The target database is `sample_mflix`
+- FerretDB is running at `mongodb://<username>:<password>@localhost:27017`
+
+Set the following environment variables for the source and destination connection strings and run dsync:
+
+```sh
+export MDB_SRC='mongodb://localhost:27018/sample_mflix'
```

Comment:
Reading the documentation, I don't find where this is documented. Does this work?

Do you want to only migrate one database? I see this https://docs.adiom.io/basics/features#namespace-filtering

Reply:
Yes it does work and also keeps the terminal open for any other updates. DSync accepts both source and destination variables. For the database, it should be all data (similar to the doc below):

https://docs.adiom.io/getting-started/quickstart/from-cosmos-db-to-mongodb-1

Reply:
I meant `sample_mflix`  in `export MDB_SRC='mongodb://localhost:27018/sample_mflix'`. The example you've shown is `export MDB_SRC='mongodb://localhost:27017' ` which doesn't have a default database name.

What I'm asking is, what is your intention of using the default database like how you did? Does it copy only one database? :thinking:

Reply:
Okay. That was a mistake I made with using `sample_mflix`. Despite that, it still copies all databases actually and syncs updates afterwards. That's why I removed the specific database_name